src/NetCDFIO.jl

```diff
@@ -4,140 +4,132 @@
+        timeseries_grp = NC.defGroup(root_grp, "timeseries")
+        NC.defDim(timeseries_grp, "t", Inf)
+        NC.defVar(timeseries_grp, "t", FT, ("t",))
+
+        #output_timeseries = ["TODO - add timeseries output"]
+        #for var in output_timeseries
+        #    NC.defVar(timeseries_grp, var, FT, ("t",))
+        #end
     end
-    vars = Dict{String, Any}()
-    return NetCDFIO_Stats(root_grp, profiles_grp, ts_grp, output_interval, nc_filename, vars)
+
+    return NetCDFIO_Stats(root_grp, output_interval, nc_filename)
 end
 
 function open_files(self::NetCDFIO_Stats)
     self.root_grp = NC.Dataset(self.nc_filename, "a")
-    self.profiles_grp = self.root_grp.group["profiles"]
-    self.ts_grp = self.root_grp.group["timeseries"]
-    vars = self.vars
-
-    # Hack to avoid https://github.com/Alexander-Barth/NCDatasets.jl/issues/135
-    vars["profiles"] = Dict{String, Any}()
-    for k in keys(self.profiles_grp)
-        vars["profiles"][k] = self.profiles_grp[k]
-    end
-    vars["timeseries"] = Dict{String, Any}()
-    for k in keys(self.ts_grp)
-        vars["timeseries"][k] = self.ts_grp[k]
-    end
 end
 
 function close_files(self::NetCDFIO_Stats)
     close(self.root_grp)
 end
 
-function write_field(self::NetCDFIO_Stats, var_name::String, data::T, group) where {T <: AbstractArray{Float64, 1}}
-    # Hack to avoid https://github.com/Alexander-Barth/NCDatasets.jl/issues/135
-    @inbounds self.vars[group][var_name][:, end] = data
-    # Ideally, we remove self.vars and use:
-    # var = self.profiles_grp[var_name]
-    # Not sure why `end` instead of `end+1`, but `end+1` produces garbage output
-    # @inbounds var[end, :] = data :: T
+function write_profiles(self::NetCDFIO_Stats, var_name::String, data::T) where {T <: AbstractArray{Float64, 1}}
+    # _parentname = nothing is needed to avoid https://github.com/Alexander-Barth/NCDatasets.jl/issues/135
+    ncdf_var = NC.cfvariable(self.root_grp.group["profiles"], var_name, _parentname = nothing)
+    # Not sure why not end+1
+    @inbounds ncdf_var[:, end] = data
 end
 
-function write_ts(self::NetCDFIO_Stats, var_name::String, data::Float64)
-    # Hack to avoid https://github.com/Alexander-Barth/NCDatasets.jl/issues/135
-    @inbounds self.vars["timeseries"][var_name][end] = data::Float64
-    # Ideally, we remove self.vars and use:
-    # var = self.ts_grp[var_name]
-    # @inbounds var[end+1] = data :: Float64
+function write_timeseries(self::NetCDFIO_Stats, var_name::String, data::Float64)
+    # _parentname = nothing is needed to avoid https://github.com/Alexander-Barth/NCDatasets.jl/issues/135
+    ncdf_var = NC.cfvariable(self.root_grp.group["timeseries"], var_name, _parentname = nothing)
+    # Not sure why not end+1
+    @inbounds ncdf_var[end] = data
 end
 
 function write_simulation_time(self::NetCDFIO_Stats, t::Float64)
     # # Write to profiles group
-    profile_t = self.profiles_grp["t"]
+    profile_t = self.root_grp.group["profiles"]["t"]
```

Comment:
You no longer need this method to depend on `NetCDFIO_Stats`, you could write as

```julia
function write_simulation_time(ds, t::FT) where {FT}
    profile_t = ds.group["profiles"]["t"]
    @inbounds profile_t[end + 1] = t::Float64
...
end
```

Also, if this is only called once, then it may be simpler to just inline this function.

The same pattern can be applied to the other functions (no need to depend on `NetCDFIO_Stats `).